src/realm/util/thread.cpp

```diff
@@ -95,6 +95,25 @@ void Thread::set_name(const std::string& name)
     int r = pthread_setname_np(name.data());
     if (REALM_UNLIKELY(r != 0))
         throw std::system_error(r, std::system_category(), "pthread_setname_np() failed");
+#elif defined(_WIN32)
+    // SetThreadDescription was only introduced in Windows 10 1607, so we need to dynamically load it.
+    HRESULT(WINAPI * SETTHREADDESCRIPTION)(HANDLE hThread, PCWSTR threadDescription) = nullptr;
+#if !defined(NTDDI_WIN10_RS1) || NTDDI_VERSION < NTDDI_WIN10_RS1
+    auto proc_address = GetProcAddress(GetModuleHandle(L"Kernel32.dll"), "SetThreadDescription");
```

Comment:
```suggestion
    static const auto proc_address = GetProcAddress(GetModuleHandle(L"Kernel32.dll"), "SetThreadDescription");
```

No reason to resolve this function on every call.